junit5/src/main/java/sunstone/core/AzureArmTemplateCloudDeploymentManager.java

```diff
@@ -0,0 +1,139 @@
+package sunstone.core;
+
+
+import com.azure.core.management.Region;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.resources.models.DeploymentMode;
+import com.azure.resourcemanager.resources.models.ResourceGroups;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.slf4j.Logger;
+import org.wildfly.extras.sunstone.api.impl.ObjectProperties;
+import org.wildfly.extras.sunstone.api.impl.ObjectType;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+
+/**
+ * Purpose: the class handles Azure template - deploy and undeploy the template to and from a stack.
+ *
+ * Used by {@link SunstoneCloudDeploy}. Deploys to a group defined in Sunstone.properties (which is deleted as a whole later).
+ *
+ * Azure ARM client credentials are taken from Sunstone.properties. See {@link AzureUtils}.
+ */
+
+class AzureArmTemplateCloudDeploymentManager implements TemplateCloudDeploymentManager {
+    static Logger LOGGER = SunstoneJUnit5Logger.DEFAULT;
+
+    private final AzureResourceManager armManager;
+    private final Set<String> usedRG;
```

Comment:
If everything is deployed to one resource group, why do we need collection to hold it?

I just want to make sure if I understand concept.